projects/hostap/build.sh

```diff
@@ -0,0 +1,138 @@
+
+cd 'tests'
+
+# Make seed corpora
+(
+  # Need clean environment for building test-tls used to create seed corpus.
+  unset CC
+  unset CXX
+  unset CFLAGS
+  unset LDFLAGS
+
+  make clean
+  make test-tls
+
+  for x in client server; do
+    ./test-tls $x write "${WORK}/test-tls-${x}.msg"
+    (cd "$WORK" && zip "${OUT}/test-tls-${x}-read_seed_corpus.zip" test-tls-${x}.msg)
+  done
+
+  (
+    cd p2p-fuzzer
+    zip "${OUT}/p2p-fuzzer-proberesp_seed_corpus.zip" proberesp*.dat
+    zip "${OUT}/p2p-fuzzer-action_seed_corpus.zip" go*.dat inv*.dat p2ps*.dat
+  )
+
+  (cd eapol-fuzzer && zip "${OUT}/eapol-fuzzer_seed_corpus.zip" *.dat)
+  (cd ap-mgmt-fuzzer && zip "${OUT}/ap-mgmt-fuzzer_seed_corpus.zip" multi.dat)
+  (cd wnm-fuzzer && zip "${OUT}/wnm-fuzzer_seed_corpus.zip" *.dat)
+
+  echo '{"a":[[]],"b":1,"c":"q","d":{"e":[{}]}}' > "${WORK}/test.json"
+  (cd "$WORK" && zip "${OUT}/test-json_seed_corpus.zip" *.json)
+
+  # TODO: test-x509
+)
+
+
+make clean
+export LDO=$CXX
+export LDFLAGS="$CFLAGS $LIB_FUZZING_ENGINE"
+export CFLAGS="$CFLAGS -DTEST_LIBFUZZER"
+
+# libFuzzer native targets (enabled via TEST_LIBFUZZER) ------------------
+
+for target in json x509; do
+  make test-${target} TEST_FUZZ=y
+  cp -v "test-${target}" "${OUT}/"
+done
+
+# AFL compatible targets --------------------------------------------------
+
+export CFLAGS="$CFLAGS -Dmain=fuzzer_main"
+
+(
+  export OBJS="../libfuzzer_entry.o"
+
+  # ap-mgmt-fuzzer
+  make clean
+  CFLAGS="$CFLAGS -DEXTRA_ARGS='\"-m\",'" \
+    make -C "ap-mgmt-fuzzer"
+  cp -v "ap-mgmt-fuzzer/ap-mgmt-fuzzer" "${OUT}/"
+
+  # wnm-fuzzer
+  rm -v "libfuzzer_entry.o"
+  make -C "wnm-fuzzer"
+  cp -v "wnm-fuzzer/wnm-fuzzer" "${OUT}/"
+)
+
+# The below Makefiles do not honor OBJS.
+recompile_libfuzzer_entry() {
+  rm -vf "libfuzzer_entry.o"
+  $CC $CFLAGS -c -o "libfuzzer_entry.o" "libfuzzer_entry.c"
+}
+
+# test-tls variants
+(
+  export LDFLAGS="$LDFLAGS libfuzzer_entry.o"
+  make clean
+
+  CFLAGS="$CFLAGS -DEXTRA_ARGS=\"server\",\"read\"," \
+    recompile_libfuzzer_entry
+  make test-tls TEST_FUZZ=y
+  cp -v "test-tls" "${OUT}/test-tls-server-read"
+
+  CFLAGS="$CFLAGS -DEXTRA_ARGS=\"client\",\"read\"," \
+    recompile_libfuzzer_entry
+  make test-tls TEST_FUZZ=y
+  cp -v "test-tls" "${OUT}/test-tls-client-read"
+)
+
+(
+  export LIBS="../libfuzzer_entry.o"
+
+  # eapol-fuzzer
+  make -C "eapol-fuzzer" clean
+  recompile_libfuzzer_entry
+  make -C "eapol-fuzzer"
+  cp -v "eapol-fuzzer/eapol-fuzzer" "${OUT}/"
+
+  # p2p-fuzzer variants
+  make -C "p2p-fuzzer" clean
+  CFLAGS="$CFLAGS -DEXTRA_ARGS=\"action\"," \
+    recompile_libfuzzer_entry
+  make -C "p2p-fuzzer"
+  cp -v "p2p-fuzzer/p2p-fuzzer" "${OUT}/p2p-fuzzer-action"
+  CFLAGS="$CFLAGS -DEXTRA_ARGS=\"proberesp\"," \
+    recompile_libfuzzer_entry
+  make -C "p2p-fuzzer"
+  cp -v "p2p-fuzzer/p2p-fuzzer" "${OUT}/p2p-fuzzer-proberesp"
+)
+
+# Copy required data.
+cp -a "hwsim" "${OUT}/"
+
+# TODO: Fix leaks and remove detect_leaks.
```

Comment:
+1, it's recommended to suppress noisy logging by compile time flags, if possible.

Reply:
Changed to supress debug output via compile time flag.